city_scrapers/spiders/il_elections.py

```diff
@@ -0,0 +1,148 @@
+            self.addresses[addr_name.split()[0]] = address
+
+    def _parse_agenda(self, response):
+        """Parse board agenda."""
+        meetings = response.css("#ContentPlaceHolder1_gvAgenda > tr")[1:-1]
+        for item in meetings:
+            start = self._parse_start(item)
+            location = self._parse_location(item)
+            meeting = Meeting(
+                title=self._parse_title(item),
+                description=self._parse_description(item),
+                classification=self._parse_classification(item),
+                start=start,
+                end=self._parse_end(item),
+                all_day=self._parse_all_day(item),
+                time_notes=self._parse_time_notes(item),
+                location=location,
+                links=self._parse_links(item, response, start),
+                source=self._parse_source(response),
+            )
+
+            meeting["status"] = self._get_status(meeting)
+            meeting["id"] = self._get_id(meeting)
+
+            yield meeting
+
+    def _parse_title(self, item):
+        """Parse or generate meeting title."""
+        return "Illinois State Board of Elections"
+
+    def _parse_description(self, item):
+        """Parse or generate meeting description."""
+        desc = item.css("td")[3].css("::text").extract_first().strip()
+        return desc
+
+    def _parse_classification(self, item):
+        """Parse or generate classification from allowed options."""
+        return BOARD
+
+    def _parse_start(self, item):
+        """Parse start datetime as a naive datetime object."""
+        date = item.css("td")[0].css("::text").extract_first().strip()
+        time = item.css("td")[1].css("::text").extract_first().strip().replace(".", "")
+        dt = date + " " + time
+        parsed_dt = datetime.strptime(dt, "%a, %B %d, %Y %I:%M %p")
+        return parsed_dt
+
+    def _parse_end(self, item):
+        """Parse end datetime as a naive datetime object. Added by pipeline if None"""
+        return None
+
+    def _parse_time_notes(self, item):
+        """Parse any additional notes on the timing of the meeting"""
+        return ""
+
+    def _parse_all_day(self, item):
+        """Parse or generate all-day status. Defaults to False."""
+        return False
+
+    def _parse_location(self, item):
+        """Parse or generate location."""
+        name = item.css("td")[2].css("::text").extract_first().strip()
+        address = self.addresses.get(name, "")
+
+        return {
+            "address": address,
+            "name": name,
+        }
+
+    def _parse_links(self, item, response, start):
+        """Parse or generate links."""
+        links = []
+        href = item.css("td")[0].css("::attr(href)").extract_first()
+        if href:
+            href = response.urljoin(href)
+            title = "Agenda"
+        else:
+            href = ""
+            title = ""
+        links.append({"href": href, "title": title})
```

Comment:
We shouldn't have empty  link dicts in the list, so any empty values should be ignored